### PR TITLE
[10.0 ][MIG] account_invoice_analytic_search

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Invoice Analytic Search
=======================

Organizations often require to quickly find the invoices associated to a project or to an analytic account, searching by it's code, name or project/account manager. This module introduces the possibility search
customer or supplier invoices by analytic account or by project manager.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

File: account_invoice_analytic_search/README.rst
Comment (LoisRForgeFlow) [inline account_invoice_analytic_search/README.rst]: Please fill the template

File: account_invoice_analytic_search/__manifest__.py
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/__manifest__.py]: IMHO this is reserved to authors. Contributors who migrate should be in the contributors section in the readme.

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

File: account_invoice_analytic_search/models/account_invoice_line.py
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/models/account_invoice_line.py]: partner_id in analytic is used for the customer not the project_manager in the project. IMHO we should  user this field instead https://github.com/OCA/OCB/blob/10.0/addons/analytic/models/analytic_account.py#L102

**Comment by nikul-serpentcs** [comment]
@aheficent Maybe you are referring to the wrong class, IMHO the related field is for account analytic account not for the line, can you please check [here](https://github.com/OCA/OCB/blob/10.0/addons/analytic/models/analytic_account.py#L54)

**Comment by AaronHForgeFlow** [comment]
@nikul-serpentcs I know what you mean. User_id is neither the project manager. So then, we have two options:
1. Remove the feature of searching by project manager
2 .Make this module depends on this one https://github.com/OCA/project/pull/307/

I would follow option 1 here. Then, another module may be created to cover searching invoices by project manager.

**Comment by nikul-serpentcs** [comment]
@aheficent I did remove the feature for searching by a project manager, Could you please check it.

**Comment by AaronHForgeFlow** [comment]
user_id is the project manager in the project not in the analytic account. I'll review

File: account_invoice_analytic_search/README.rst
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/README.rst]: This is not possible anymore, can you remove it from the readme?

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please look it.

File: account_invoice_analytic_search/models/account_invoice_line.py
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/models/account_invoice_line.py]: What is this for?

File: account_invoice_analytic_search/models/account_invoice.py
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/models/account_invoice.py]: This is not necessary anymore

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Tested as follow:
- created an invoice with one line containing an analytic account.
- searched invoices by the analytic name. No results 
found.
- searched invoices by the analytic code. No results found.
Same results for customer invoices and  supplier invoices. Can you add some tests please?

**Comment by nikul-serpentcs** [comment]
@aheficent Added a Unit test case, Could you please look it.

File: account_invoice_analytic_search/README.rst
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/README.rst]: Sorry for the misunderstanding. What i meant before was that now it's not possible to search by project manager. The only part I asked you to remove was the one related to the project manager

File: account_invoice_analytic_search/models/account_invoice.py
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/models/account_invoice.py]: You should delete this also

File: account_invoice_analytic_search/models/account_invoice.py
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/models/account_invoice.py]: You are returning the domain here for the invoice lines not the invoice.

File: account_invoice_analytic_search/tests/test_account_invoice_analytic_search.py
Comment (AaronHForgeFlow) [inline account_invoice_analytic_search/tests/test_account_invoice_analytic_search.py]: You are comparing the list of invoice_line_ids here not the name with the code. That's why it is not failing

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Still not working

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please look it.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: LGTM now. Thank you for attending the comments

File: account_invoice_analytic_search/models/account_invoice.py
Comment (rousseldenis) [inline account_invoice_analytic_search/models/account_invoice.py:14]: @nikul-serpentcs You'd better depends on 'invoice_line_ids.account_analytic_id' as all recordsets will be fetched with these before entering loop.

File: account_invoice_analytic_search/models/account_invoice.py
Comment (rousseldenis) [inline account_invoice_analytic_search/models/account_invoice.py:25]: @nikul-serpentcs To avoid a search (and so an sql query), return that domain.

And to be complete as avoiding one more search do just this in this function: 

```python
return [('invoice_line_ids.account_analytic_id', operator, value)]
```

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by AaronHForgeFlow** [comment]
comments attended here: #366 @rousseldenis @nikul-serpentcs

**Comment by feketemihai** [comment]
@OCA/accounting-maintainers We should close this one, and merge #366 since it's a continuation and from the 2 PR's we have 3 approvals.

**Pull request closed without merging**
